include/circt/Dialect/Moore/MooreOps.td

```diff
@@ -73,6 +73,69 @@ def InstanceOp : MooreOp<"instance", [
   }];
 }
 
+def Initial: I32EnumAttrCase<"Initial", 0, "initial">;
+def Final: I32EnumAttrCase<"Final", 1, "final">;
+def Always: I32EnumAttrCase<"Always", 2, "always">;
+def AlwaysComb: I32EnumAttrCase<"AlwaysComb", 3, "always_comb">;
+def AlwaysLatch: I32EnumAttrCase<"AlwaysLatch", 4, "always_latch">;
+def AlwaysFF: I32EnumAttrCase<"AlwaysFF", 5, "always_ff">;
+
+def ProcedureKindAttr: I32EnumAttr<"ProcedureKind", "Procedure kind",
+            [Initial, Final, Always, AlwaysComb, AlwaysLatch, AlwaysFF]>{
+  let cppNamespace = "circt::moore";
+}
+
+def ProcedureOp : MooreOp<"procedure", [
```

Comment:
I think one difference from SV dialect is that all procedural operations is implemented in a single operation. I don't have a strong opinion but if we are going to implement sensitivity list it might make sense to split into several operations.

Reply:
They are all identical, except for when they are executed during simulation. But there are no sensitivity lists or anything the like.